### ISSUE TITLE
PowerPC 64 little-endian crash with Ubuntu 22.04 and gcc 11

### ISSUE THREAD

**Issue by mtl1979** [zlib-ng/zlib-ng#1378]
When compiled with gcc 11.3.0, all tests fail or crash if qemu run with POWER8 as cpu, none fail if run with POWER9 as cpu..
When compiled with gcc 9.4.0, only `gtest_zlib` crash.

This suggests qemu in Ubuntu 22.04 can't run binaries compiled with either gcc 9.4.0 or 11.3.0. It emulates some instruction incorrectly for POWER8.

See: https://gitlab.com/qemu-project/qemu/-/issues/1361

**Comment by pkubaj** [comment]
https://www.phoronix.com/news/Ubuntu-22.04-LTS-POWER9
```
Starting with Ubuntu 22.04 LTS, POWER9 and POWER10 processors are supported.

The support for POWER8 ends with Ubuntu 21.10, respectively Ubuntu 20.04 LTS, which is still supported for several years.
```
https://ubuntu.com/download/server/power

**Comment by mtl1979** [comment]
As qemu is still broken in Ubuntu 22.04, we don't really need to worry about changing the default (minimum supported) target to POWER9... We're supposed to support older Ubuntu releases, which will still run with POWER8 hardware.

I will keep this issue open for tracking if fixed qemu is added to Ubuntu 22.04 repositories... Then we can re-evaluate if we want to add CI run for POWER9 under Ubuntu 22.04.

**Comment by ccawley2011** [comment]
> Then we can re-evaluate if we want to add CI run for POWER9 under Ubuntu 22.04.

Wouldn't it be necessary to add CI runs for POWER9 anyway so that `compare256_power9` is included in the test coverage?

**Comment by mtl1979** [comment]
@ccawley2011 If qemu is buggy in Ubuntu 22.04 (it is known to be buggy since 5.0 series), it can result in incorrect assumption that there is a bug in our code... I've tried to research what exactly is the core reason of the bug, but nobody online really knows it yet. Only thing I know is that it throws page fault when reading memory region that clearly is mapped to a dependency, not zlib-ng itself. Unless we can find POWER9 CI runners that don't use Docker and qemu, we have to wait until the bug is investigated and fixed.

Like I said earlier, it's still possible to use older Ubuntu release (which have qemu 4.0), but GitHub has had long queues with older releases, so we shouldn't try to use them too much.

**Comment by ccawley2011** [comment]
I take it that means the issue is something other than using shared libraries targeting POWER9 when emulating POWER8.

Also, if QEMU 5.x is known to be buggy, does that mean that the big endian and configure workflows need to be downgraded as well?

**Comment by mtl1979** [comment]
@ccawley2011 qemu has had a long history of bugs that have affected only some of the emulated processors and in some cases only some of the host processors. This heavily suggests not all architecture pairs use same buggy code paths. As much I'd like to downgrade all "ubuntu-latest" runners to "ubuntu-20.04", it would likely cause days worth of delays in every single pull request. Recently I have received last notifications for a pull request made 3 days prior.

**Comment by nmoinvaz** [comment]
@mtl1979 that was already merged in as #1394. Perhaps we can close this issue?

**Comment by mtl1979** [comment]
@nmoinvaz Or convert to a discussion... I still have quite limited privileges on this repository...
